charts/internal/seed-controlplane/charts/csi-driver-filestore-controller/templates/csi-driver-filestore-controller.yaml

```diff
@@ -0,0 +1,266 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: csi-driver-filestore-controller
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: csi
+    role: controller
+    high-availability-config.resources.gardener.cloud/type: controller
+spec:
+  replicas: {{ .Values.replicas }}
+  revisionHistoryLimit: 0
+  selector:
+    matchLabels:
+      app: csi
+      role: controller
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      annotations:
+        checksum/configmap-csi-driver-controller: {{ include (print $.Template.BasePath "/config.yaml") . | sha256sum }}
```

Comment:
Curiously, we don't restart the container if the cloudprovider credentials are changed. Maybe that's a nice addition (also to our normal CSI driver). I don't remember if there is any reason not to do it like that.